kuma/core/decorators.py

```diff
@@ -1,19 +1,41 @@
-import re
 from functools import wraps, partial
+import re
 
 from django.conf import settings
-from django.shortcuts import redirect, render
 from django.contrib.auth import REDIRECT_FIELD_NAME
 from django.contrib.auth.decorators import user_passes_test
 from django.core.exceptions import PermissionDenied
 from django.http import HttpResponseForbidden, HttpResponseRedirect
-from django.utils.decorators import available_attrs
+from django.shortcuts import redirect, render
 from django.utils.http import urlquote
+from django.utils.decorators import available_attrs
+from django.views.decorators.cache import cache_control
 
 from .jobs import BannedIPsJob
 from .urlresolvers import reverse
 
 
+def shared_cache_control(func=None, **kwargs):
```

Comment:
Nit: The name "shared cache" confused me a bit. Is the intent is to save a copy in the proxy, but to tell the browser to not cache the results? If so, a better name doesn't come to mind yet...

Reply:
Yes, that was the intent. I'm happy to use whatever name you'd prefer, as I had a difficult time coming up with something better.

Reply:
``proxy_cache_yes_browser_cache_no_with_overrides`` is my current best choice, which is so awful. Let's merge with this name and revisit in a few weeks when we're smarter.